tests/base/test_directives.py

```diff
@@ -103,22 +103,78 @@ def test_validation_in_inversion(self):
             # validation should happen and this will fail
             # (IRLS needs to be before update_Jacobi)
             inv = inversion.BaseInversion(
-                self.invProb, directiveList=[betaest, update_Jacobi, IRLS]
+                invProb, directiveList=[betaest, update_Jacobi, IRLS]
             )
 
         with self.assertRaises(AssertionError):
             # validation should happen and this will fail
             # (sensitivity_weights needs to be before betaest)
             inv = inversion.BaseInversion(
-                self.invProb, directiveList=[betaest, sensitivity_weights]
+                invProb, directiveList=[betaest, sensitivity_weights]
             )
 
         with self.assertRaises(AssertionError):
             # validation should happen and this will fail
             # (sensitivity_weights needs to be before update_Jacobi)
-            inv = inversion.BaseInversion(self.invProb)
+            inv = inversion.BaseInversion(invProb)
             inv.directiveList = [update_Jacobi, sensitivity_weights]
 
+    def test_sensitivity_weighting(self):
+
+        tests_list = [
```

Comment:
I would recommend making these as differently ran tests, just in case the first option fails it will still test the second. You can still re-use most of the code, just give these inputs to the internal test func.

Reply:
Done